.github/workflows/tests-contributors.yml

```diff
@@ -37,30 +37,77 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Generate matrix for test jobs
+        id: generate_matrix
+        run: |
+          FEDVERS=34
+          TARGET_BRANCH=${{ fromJson(steps.pr_api.outputs.data).base.ref }}
+
+          JSON_MASTER_ELN='{"release": "master", "target-branch": "master", "ci-tag": "master"}, {"release": "eln", "target-branch": "master", "ci-tag": "eln", "build-args": "--build-arg=image=quay.io/fedoraci/fedora:eln-x86_64"}'
+          JSON_DEVEL='{"release": "f'$FEDVERS'-devel", "target-branch": "f'$FEDVERS'-devel", "ci-tag": "f'$FEDVERS'-devel"}'
+          JSON_RELEASE='{"release": "f'$FEDVERS'-release", "target-branch": "f'$FEDVERS'-release", "ci-tag": "f'$FEDVERS'-release"}'
+
+          REF_DEVEL='f'$FEDVERS'-devel'
+          REF_RELEASE='f'$FEDVERS'-release'
+
+          # Always order things so that first array element is the current branch, because... wait for it...
+          if [[ "$TARGET_BRANCH" == "$REF_DEVEL" ]] ; then
+            MERGED='{"include": ['$JSON_DEVEL', '$JSON_RELEASE', '$JSON_MASTER_ELN']}' ;
+          elif [[ "$TARGET_BRANCH" == "$REF_RELEASE" ]] ; then
+            MERGED='{"include": ['$JSON_RELEASE']}' ;
+          elif [[ "$TARGET_BRANCH" == "master" ]] ; then
+            MERGED='{"include": ['$JSON_MASTER_ELN']}' ;
+          else
+            OUT_JSON_FULL=''
+            OUT_JSON_PRIMARY=''
+            OUT_JSON_SECONDARY=''
+          fi
+
+          if [[ -z $OUT_JSON_FULL ]] ; then
+            # ...here "release" in first array element is replaced with empty string to make the displayed name always the same.
+            OUT_JSON_FULL=$(echo "$MERGED" | jq -c '.include[0].release |= ""')
+            # Turn "release" values into an array to be used as the primary key for matrix. Do not use the whole JSON directly, it generates FUGLY job names.
+            OUT_JSON_PRIMARY=$(echo "$OUT_JSON_FULL" | jq -c '[.include[].release]')
+            # To use the whole JSON definition, pass the array to include to get what include: does.
+            OUT_JSON_SECONDARY=$(echo "$OUT_JSON_FULL" | jq -c '.include')
+          fi
+
+          # Pretty-print the results for debug and set the outputs
+          echo "Generated JSONs are:"
+          echo "$OUT_JSON_FULL" | jq '.'
+          echo "$OUT_JSON_PRIMARY" | jq '.'
+          echo "$OUT_JSON_SECONDARY" | jq '.'
+          echo "::set-output name=matrix-full::$OUT_JSON_FULL"
+          echo "::set-output name=matrix-primary::$OUT_JSON_PRIMARY"
+          echo "::set-output name=matrix-secondary::$OUT_JSON_SECONDARY"
+
     outputs:
       allowed_user: ${{ steps.check_user_perm.outputs.allowed_user }}
       base_ref: ${{ fromJson(steps.pr_api.outputs.data).base.ref }}
       sha: ${{ fromJson(steps.pr_api.outputs.data).head.sha }}
+      matrix-full: ${{ steps.generate_matrix.outputs.matrix-full }}
```

Comment:
If I see it correctly this is not used anywhere. Let's simplify the logic and everything here and have just one matrix not three.

Reply:
Agreed. Done.